src/Glpi/Asset/Capacity/HasImpactCapacity.php

```diff
@@ -0,0 +1,105 @@
+<?php
+
+/**
+ * ---------------------------------------------------------------------
+ *
+ * GLPI - Gestionnaire Libre de Parc Informatique
+ *
+ * http://glpi-project.org
+ *
+ * @copyright 2015-2024 Teclib' and contributors.
+ * @copyright 2003-2014 by the INDEPNET Development Team.
+ * @licence   https://www.gnu.org/licenses/gpl-3.0.html
+ *
+ * ---------------------------------------------------------------------
+ *
+ * LICENSE
+ *
+ * This file is part of GLPI.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ * ---------------------------------------------------------------------
+ */
+
+namespace Glpi\Asset\Capacity;
+
+use CommonGLPI;
+use Config;
+use Impact;
+use ImpactItem;
+
+class HasImpactCapacity extends AbstractCapacity
+{
+    public function getLabel(): string
+    {
+        return Impact::getTypeName(1);
+    }
+
+    public function getIcon(): string
+    {
+        return Impact::getIcon();
+    }
+
+    public function isUsed(string $classname): bool
+    {
+        //NOTE: This doesn't consider cases where assets exist inside graphs on other asset types. Maybe there is no good solution since it involves scanning every graph.
+        return parent::isUsed($classname)
+            && countElementsInTable(ImpactItem::getTable(), ['itemtype' => $classname]) > 0;
```

Comment:
What do you mean by "scanning" every graph ?

Couldn't we just count items in `glpi_impactrelations` where `itemtype_source` or `itemtype_impacted` is `$classname` ?

Reply:
Oh. Yes, we can. I thought for sure all the data was stored in JSON, but I guess it is just information for the designer and not relevant.